### PR TITLE
Create Strassen’s Matrix Multiplication

### PR DESCRIPTION
**Pull request by YashBhimani06**
Whole code for algorithm.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by YashBhimani06** [comment]
New algorithm for repo

File: divide_and_conquer/Strassen’s Matrix Multiplication.cpp [hunk 1]
```diff
@@ -0,0 +1,182 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+#define ROW_1 4
+#define COL_1 4
+
+#define ROW_2 4
+#define COL_2 4
+
```
Comment (martin-sos) [inline divide_and_conquer/Strassen’s Matrix Multiplication.cpp:9]: Would it improve readability to use something like `using Matrix = vector<vector<int>>;` and replace all occurences of  `vector<vector<int>>`  accordingly?

File: divide_and_conquer/Strassen’s Matrix Multiplication.cpp [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+				= matrix_A[i][j]
+				+ (multiplier * matrix_B[i][j]);
+	return matrix_A;
+}
+
+vector<vector<int> >
+multiply_matrix(vector<vector<int> > matrix_A,
+				vector<vector<int> > matrix_B)
+{
+	int col_1 = matrix_A[0].size();
```
Comment (martin-sos) [inline divide_and_conquer/Strassen’s Matrix Multiplication.cpp:43]: size returns a size_t

File: divide_and_conquer/Strassen’s Matrix Multiplication.cpp [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+		result_matrix_00.clear();
+		result_matrix_01.clear();
+		result_matrix_10.clear();
+		result_matrix_11.clear();
+	}
+	return result_matrix;
+}
+
+int main()
+{
```
Comment (martin-sos) [inline divide_and_conquer/Strassen’s Matrix Multiplication.cpp:158]: There is no test for correctness of the multiplication.

Comment (martin-sos) [reply divide_and_conquer/Strassen’s Matrix Multiplication.cpp:158]: ...a nice test implementation is here: https://martin-thoma.com/matrix-multiplication-python-java-cpp/#The_Tests

**Pull request closed without merging**